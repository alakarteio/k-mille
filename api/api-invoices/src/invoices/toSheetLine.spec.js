/* eslint-env jest */
const toSheetLine = require('./toSheetLine')

describe('toSheetLine', () => {
  const baseInvoice = {
    client: {
      name: 'client-name',
      city: 'client-city',
      address: 'client-address',
    },
    dates: {
      end: 1522620000000,
      print: 1531472327262,
    },
    lines: [
      {
        title: '2',
        nb: '10',
        pricePerUnit: '100',
      },
    ],
    id: '00006',
    timetable: [],
    fileId: '1xm33aSYp62yAxht0a5f-0kw7hTX_z3I5',
    ranges: [
      7,
    ],
  }

  const baseInvoiceWithTimetable = Object.assign(
    {},
    baseInvoice,
    {
      timetable: [
        {
          date: '2018-03-01',
          price: '133',
        },
        {
          date: '2018-04-01',
          price: '123',
          paid: true,
        },
      ],
      ranges: [
        7,
        8,
      ],
    },
  )

  it('should parse a simple invoice -modification-', () => {
    const invoice = Object.assign({}, baseInvoice, { paid: true })

    expect(toSheetLine(invoice)).toMatchSnapshot()
  })

  it('should parse a timetable invoice -modification-', () => {
    const invoice = Object.assign({}, baseInvoiceWithTimetable)

    expect(toSheetLine(invoice)).toMatchSnapshot()
  })

  it('should parse a timetable invoice -creation-', () => {
    const invoice = Object.assign({}, baseInvoiceWithTimetable)
    delete invoice.ranges

    expect(toSheetLine(invoice)).toMatchSnapshot()
  })

  it('should parse a simple invoice -creation-', () => {
    const invoice = Object.assign({}, baseInvoice)
    delete invoice.ranges

    expect(toSheetLine(invoice)).toMatchSnapshot()
  })
})
