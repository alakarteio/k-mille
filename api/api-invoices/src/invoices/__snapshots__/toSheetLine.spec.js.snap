// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toSheetLine should parse a simple invoice -creation- 1`] = `
Array [
  Array [
    "{\\"client\\":{\\"name\\":\\"client-name\\",\\"city\\":\\"client-city\\",\\"address\\":\\"client-address\\"},\\"dates\\":{\\"end\\":1522620000000,\\"print\\":1531472327262},\\"lines\\":[{\\"title\\":\\"2\\",\\"nb\\":\\"10\\",\\"pricePerUnit\\":\\"100\\"}],\\"id\\":\\"00006\\",\\"timetable\\":[],\\"fileId\\":\\"1xm33aSYp62yAxht0a5f-0kw7hTX_z3I5\\"}",
    "client-name",
    6,
    "2018-07-13",
    1000,
    "non",
    "https://drive.google.com/file/d/1xm33aSYp62yAxht0a5f-0kw7hTX_z3I5/view",
  ],
]
`;

exports[`toSheetLine should parse a simple invoice -modification- 1`] = `
Array [
  Array [
    "{\\"client\\":{\\"name\\":\\"client-name\\",\\"city\\":\\"client-city\\",\\"address\\":\\"client-address\\"},\\"dates\\":{\\"end\\":1522620000000,\\"print\\":1531472327262},\\"lines\\":[{\\"title\\":\\"2\\",\\"nb\\":\\"10\\",\\"pricePerUnit\\":\\"100\\"}],\\"id\\":\\"00006\\",\\"timetable\\":[],\\"fileId\\":\\"1xm33aSYp62yAxht0a5f-0kw7hTX_z3I5\\",\\"paid\\":true}",
    "client-name",
    6,
    "2018-07-13",
    1000,
    "oui",
    "https://drive.google.com/file/d/1xm33aSYp62yAxht0a5f-0kw7hTX_z3I5/view",
  ],
]
`;

exports[`toSheetLine should parse a timetable invoice -creation- 1`] = `
Array [
  Array [
    "{\\"client\\":{\\"name\\":\\"client-name\\",\\"city\\":\\"client-city\\",\\"address\\":\\"client-address\\"},\\"dates\\":{\\"end\\":1522620000000,\\"print\\":1531472327262},\\"lines\\":[{\\"title\\":\\"2\\",\\"nb\\":\\"10\\",\\"pricePerUnit\\":\\"100\\"}],\\"id\\":\\"00006\\",\\"timetable\\":[{\\"date\\":\\"2018-03-01\\",\\"price\\":\\"133\\"},{\\"date\\":\\"2018-04-01\\",\\"price\\":\\"123\\",\\"paid\\":true}],\\"fileId\\":\\"1xm33aSYp62yAxht0a5f-0kw7hTX_z3I5\\"}",
    "client-name",
    6,
    "2018-03-01",
    133,
    "non",
    "https://drive.google.com/file/d/1xm33aSYp62yAxht0a5f-0kw7hTX_z3I5/view",
  ],
  Array [
    "{\\"client\\":{\\"name\\":\\"client-name\\",\\"city\\":\\"client-city\\",\\"address\\":\\"client-address\\"},\\"dates\\":{\\"end\\":1522620000000,\\"print\\":1531472327262},\\"lines\\":[{\\"title\\":\\"2\\",\\"nb\\":\\"10\\",\\"pricePerUnit\\":\\"100\\"}],\\"id\\":\\"00006\\",\\"timetable\\":[{\\"date\\":\\"2018-03-01\\",\\"price\\":\\"133\\"},{\\"date\\":\\"2018-04-01\\",\\"price\\":\\"123\\",\\"paid\\":true}],\\"fileId\\":\\"1xm33aSYp62yAxht0a5f-0kw7hTX_z3I5\\"}",
    "client-name",
    6,
    "2018-04-01",
    123,
    "oui",
    "https://drive.google.com/file/d/1xm33aSYp62yAxht0a5f-0kw7hTX_z3I5/view",
  ],
]
`;

exports[`toSheetLine should parse a timetable invoice -modification- 1`] = `
Array [
  Array [
    "{\\"client\\":{\\"name\\":\\"client-name\\",\\"city\\":\\"client-city\\",\\"address\\":\\"client-address\\"},\\"dates\\":{\\"end\\":1522620000000,\\"print\\":1531472327262},\\"lines\\":[{\\"title\\":\\"2\\",\\"nb\\":\\"10\\",\\"pricePerUnit\\":\\"100\\"}],\\"id\\":\\"00006\\",\\"timetable\\":[{\\"date\\":\\"2018-03-01\\",\\"price\\":\\"133\\"},{\\"date\\":\\"2018-04-01\\",\\"price\\":\\"123\\",\\"paid\\":true}],\\"fileId\\":\\"1xm33aSYp62yAxht0a5f-0kw7hTX_z3I5\\"}",
    "client-name",
    6,
    "2018-03-01",
    133,
    "non",
    "https://drive.google.com/file/d/1xm33aSYp62yAxht0a5f-0kw7hTX_z3I5/view",
  ],
  Array [
    "{\\"client\\":{\\"name\\":\\"client-name\\",\\"city\\":\\"client-city\\",\\"address\\":\\"client-address\\"},\\"dates\\":{\\"end\\":1522620000000,\\"print\\":1531472327262},\\"lines\\":[{\\"title\\":\\"2\\",\\"nb\\":\\"10\\",\\"pricePerUnit\\":\\"100\\"}],\\"id\\":\\"00006\\",\\"timetable\\":[{\\"date\\":\\"2018-03-01\\",\\"price\\":\\"133\\"},{\\"date\\":\\"2018-04-01\\",\\"price\\":\\"123\\",\\"paid\\":true}],\\"fileId\\":\\"1xm33aSYp62yAxht0a5f-0kw7hTX_z3I5\\"}",
    "client-name",
    6,
    "2018-04-01",
    123,
    "oui",
    "https://drive.google.com/file/d/1xm33aSYp62yAxht0a5f-0kw7hTX_z3I5/view",
  ],
]
`;
